internal/test/e2e/e2e_test.go

```diff
@@ -103,33 +103,17 @@ func TestE2E_Kind(t *testing.T) {
 
 	// Deploy the operator.
 	t.Log("Applying operator")
-	err = kubectl.Apply("--kustomize", filepath.Join(*fRepoRoot, "config", "rbac"))
-	require.NoError(t, err)
-	err = kubectl.Apply("--kustomize", filepath.Join(*fRepoRoot, "config", "manager"))
-	require.NoError(t, err)
+	kustomizeEdit := exec.Command("kustomize", "edit", "set", "image", fmt.Sprintf("controller=%s", operatorImage))
+	kustomizeEdit.Dir = filepath.Join(*fRepoRoot, "config", "manager")
+	out, err = kustomizeEdit.CombinedOutput()
+	require.NoError(t, err, string(out))
 
-	// Patch in our test image to the operator.
-	err = kubectl.Patch(
-		"--kustomize", filepath.Join(*fRepoRoot, "config", "manager"),
-		"--type", "json",
-		"--patch", fmt.Sprintf(`[
-			{
-				"op": "replace", 
-				"path": "/spec/template/spec/containers/0/image", 
-				"value": %q,
-			},
-			{
-				"op": "replace", 
-				"path": "/spec/template/spec/containers/0/imagePullPolicy", 
-				"value": "Never",
-			},
-		]`, operatorImage),
-	)
+	err = kubectl.Apply("--kustomize", filepath.Join(*fRepoRoot, "config", "default"))
 	require.NoError(t, err)
 
 	// Ensure the operator starts.
 	err = try.Eventually(func() error {
-		out, err := kubectl.Get("deploy", "etcd-cluster-operator", "-o=jsonpath='{.status.readyReplicas}'")
+		out, err := kubectl.Get("--namespace", "etcd-cluster-operator-system", "deploy", "etcd-cluster-operator-manager", "-o=jsonpath='{.status.readyReplicas}'")
```

Comment:
i wrote it with the patch initially to avoid
1) having a dependency on kustomize (another thing to worry about installing in CI)
2) having the test modify the state of the repo (running this now will cause a diff in config/manager/kustomize.yaml)

patching over the image is not the cleanest solution but i thought it nicer than doing this, wdyt?

Reply:
> having a dependency on kustomize (another thing to worry about installing in CI)

I did consider this, but you need it anyway for `make` to work as well call it in there.

> having the test modify the state of the repo (running this now will cause a diff in config/manager/kustomize.yaml)

Yeah I also thought about that. We *could* have a git-ignored directory under `config/test` or something that just brings in `config/default` as a base and then edits to that directory would be ignored.

The patch was causing headaches because previously you could load `config/manager` and patch that, but now without the name prefixing it doesn't apply. We could add a custom patch and apply that, not depending on the kustomzie at all?

Another option is to simply publish our controller into KIND using the default image name of `controller:latest`. Wouldn't set the `ImagePullPolicy` though.

Reply:
> I did consider this, but you need it anyway for make to work as well call it in there.

true, but its not currently a dependency of anything we run in CI

> you could load config/manager and patch that, but now without the name prefixing it doesn't apply

if the name/namespace is still predictable, we can `kubectl patch deploy $name -p ...` instead.

> Another option is to simply publish our controller into KIND using the default image name of controller:latest. Wouldn't set the ImagePullPolicy though.

unfortunately not :( using the `:latest` tag forces `ImagePullPolicy: Always`, which won't work with the way Kind loads images in to the k8s nodes.